pkg/daemon/ceph/client/upgrade.go

```diff
@@ -182,16 +182,19 @@ func OkToStop(context *clusterd.Context, namespace, deployment, daemonType, daem
 }
 
 // OkToContinue determines if it's ok to continue an upgrade
-func OkToContinue(context *clusterd.Context, namespace, deployment, daemonType, daemonName string) error {
+func OkToContinue(context *clusterd.Context, namespace, deployment, daemonType, daemonName string, continueUpgradeAfterChecksEvenIfNotClean bool) error {
 	// the mon case is handled directly in the deployment where the mon checks for quorum
 	switch daemonType {
 	case "osd":
 		if osdDoNothing(context, namespace) {
 			return nil
 		}
 		err := okToContinueOSDDaemon(context, namespace)
-		if err != nil {
+		if err != nil && !continueUpgradeAfterChecksEvenIfNotClean {
 			return errors.Wrapf(err, "failed to check if %s was ok to continue", deployment)
+		} else if err != nil && continueUpgradeAfterChecksEvenIfNotClean {
+			logger.Infof("PGs are not clean but continueUpgradeAfterChecksEvenIfNotClean is %t, so let's continue...", continueUpgradeAfterChecksEvenIfNotClean)
```

Comment:
nit: we know it is true, so no need to format the value with `%t`

Reply:
True :)